subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -810,6 +814,43 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 #endif
+#if defined(CONFIG_LOG_BACKEND_NET_USE_DHCPV4_OPTION)
+		case DHCPV4_OPTIONS_LOG_SERVER: {
+			struct sockaddr_in log_server = { 0 };
+
+			/* Log server option may present 1 or more
+			 * addresses. Each 4 bytes in length. Log
+			 * servers should be listed in order
+			 * of preference.  Hence we choose the first
+			 * and skip the rest.
+			 */
+			if (length % 4 != 0U) {
+				NET_ERR("options_log_server, bad length");
+				return false;
+			}
+
+			if (net_pkt_read(pkt, log_server.sin_addr.s4_addr, 4) < 0 ||
+			    net_pkt_skip(pkt, length - 4U) < 0) {
+				NET_ERR("options_log_server, short packet");
+				return false;
+			}
+
+			log_server.sin_family = AF_INET;
+			log_backend_net_set_ip((struct sockaddr *)&log_server);
+
+#ifdef CONFIG_LOG_BACKEND_NET_AUTOSTART
+			const struct log_backend *backend = log_backend_net_get();
+
+			if (!log_backend_is_active(backend)) {
+				log_backend_activate(backend, backend->cb->ctx);
+			}
+#endif /* CONFIG_LOG_BACKEND_NET_AUTOSTART */
```

Comment:
This does not really belong to DHCP module. IMHO we should just call a function, could be called `log_backend_net_start()`, in net log backend, which then would have the these checks in them and do what is needed.